### PR TITLE
Add support to Disco F746NG

### PR DESCRIPTION
**Pull request by ristohuhtala**
### Summary of changes 

[yes] I confirm this contribution is my own and I agree to license it with Apache 2.0.

[yes] I confirm the moderators may change the PR before merging it in.

For new board enablements only:

[yes] I confirm the board is [Mbed Enabled](https://www.mbed.com/en/about-mbed/mbed-enabled/introduction/) and passes the Mbed Enabled test set.

[yes] I confirm the contribution has been tested properly and the tests results for [TESTS](https://github.com/ARMmbed/mbed-os-example-pelion/tree/master/TESTS) are attached.

[yes] I confirm `mbed-os.lib` and `mbed-cloud-client.lib` hashes or the content in folders `mbed-os` and `mbed-cloud-client` were not modified in order to pass the tests.

- Maintainers of this repository update the Mbed OS and Client hashes.
- Please report any issues through issues in relevant repositories.

### CODE REVIEW

**Comment by iot-pdmc** [comment]
Thanks for your changes.
@ARMmbed/pdmc-integrators please review.

File: mbed_app.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
             "platform.stdio-flush-at-exit"              : true,
             "rtos.main-thread-stack-size"               : 5120,
             "update-client.storage-locations"           : 1,
-            "mbed-trace.enable"                         : null,
+            "mbed-trace.enable"                         : 1,
```
Comment (teetak01) [inline mbed_app.json:19]: revert this

**Comment by teetak01** [comment]
run-ci

**Comment by teetak01** [comment]
@ristohuhtala the WIP can be removed?

**Comment by ristohuhtala** [comment]
> @ristohuhtala the WIP can be removed?

Removed

**Comment by iot-pdmc** [comment]
# [Test run: SUCCESS](https://jenkins-client.isgtesting.com/job/mbed-os-example-pelion/659/)
***

**Test summary:**
Executed: 24
Failed: 0
Skipped: 0
Errors: 0
[Build artifacts](http://pdmc-ci.s3-website-eu-west-1.amazonaws.com/?prefix=jenkins-ci/mbed-os-example-pelion/659)

posted-by-tag: https://jenkins-client.isgtesting.com/job/mbed-os-example-pelion/

**Pull request merged**